server/schedule/checker/rule_checker.go

```diff
@@ -222,11 +223,23 @@ func (c *RuleChecker) replaceUnexpectRulePeer(region *core.RegionInfo, rf *place
 		}
 	}
 
+	var witness *metapb.Peer
+	for _, witness = range region.GetWitnesses() {
+		if witness.StoreId != peer.StoreId {
+			break
+		}
+	}
+
 	createOp := func() (*operator.Operator, error) {
 		if newLeader != nil && newLeader.GetId() != peer.GetId() {
 			return operator.CreateReplaceLeaderPeerOperator("replace-rule-"+status+"-leader-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer, newLeader)
 		}
-		return operator.CreateMovePeerOperator("replace-rule-"+status+"-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer)
+		/* TODO: enable this
```

Comment:
When will we enable this?

Reply:
After https://github.com/tikv/pd/pull/5285 is merged.